testflight_invite.py

```diff
@@ -117,25 +128,62 @@ def service_key(self):
             raise ValueError('Unable to find iTunes Connect Service key')
         return matches.group(1)
 
+    def getGroups(self):
+        text = self.readData('https://itunesconnect.apple.com/testflight/v2/providers/{teamId}/apps/{appId}/groups'.format(
+            teamId=self.contentProviderId, appId=self.appId
+        ))
+        data = json.loads(text)
+        return data['data']
+
+    def getDefaultExternalGroupId(self):
+        if not hasattr(self, 'groupData'):
+            self.groupData = self.getGroups()
+        defaultExternalGroups = [g for g in self.groupData if g['isDefaultExternalGroup']]
+        if len(defaultExternalGroups) > 0:
+            return defaultExternalGroups.pop()['id']
+        else:
+            return None
+
+    def getFirstContentProviderId(self):
+        # Sample return value
+        # => {"associatedAccounts"=>
+        #   [{"contentProvider"=>{"contentProviderId"=>11142800, "name"=>"Felix Krause", "contentProviderTypes"=>["Purple Software"]}, "roles"=>["Developer"], "lastLogin"=>1468784113000}],
+        #  "sessionToken"=>{"dsId"=>"8501011116", "contentProviderId"=>18111111, "expirationDate"=>nil, "ipAddress"=>nil},
+        text = self.readData("https://itunesconnect.apple.com/WebObjects/iTunesConnect.woa/ra/user/detail")
+        data = json.loads(text)
+        account = data['data']['associatedAccounts'][0]
+        return account['contentProvider']['contentProviderId']
+
     def login(self):
-        data = {
-            'accountName': self.itcLogin,
-            'password': self.itcPassword,
-            'rememberMe': 'false'
-        }
-        headers = {
-            'Content-Type': 'application/json',
-            'X-Requested-With': 'XMLHttpRequest',
-            'Accept': 'application/json, text/javascript'
-        }
+        if not self.loggedIn:
+            data = {
+                'accountName': self.itcLogin,
+                'password': self.itcPassword,
+                'rememberMe': 'false'
+            }
+            headers = {
+                'Content-Type': 'application/json',
+                'X-Requested-With': 'XMLHttpRequest',
+                'Accept': 'application/json, text/javascript'
+            }
 
-        loginUrl = 'https://idmsa.apple.com/appleauth/auth/signin?widgetKey=%s' % self.service_key
-        self.readData(
-            'https://idmsa.apple.com/appleauth/auth/signin?widgetKey=%s' % self.service_key,
-            data=json.dumps(data),
-            headers=headers
-        )
-        self.readData("https://itunesconnect.apple.com/WebObjects/iTunesConnect.woa/wa")
+            loginUrl = 'https://idmsa.apple.com/appleauth/auth/signin?widgetKey=%s' % self.service_key
+            self.readData(
+                'https://idmsa.apple.com/appleauth/auth/signin?widgetKey=%s' % self.service_key,
+                data=json.dumps(data),
+                headers=headers
+            )
+
+            if not hasattr(self, 'contentProviderId'):
```

Comment:
Would prefer if you always defined it in `__init__` and then just checked if it had a value here.